parity/cli/mod.rs

```diff
@@ -506,18 +506,18 @@ usage! {
 			"--jsonrpc-hosts=[HOSTS]",
 			"List of allowed Host header values. This option will validate the Host header sent by the browser, it is additional security against some attack vectors. Special options: \"all\", \"none\",.",
 
-			ARG arg_jsonrpc_threads: (usize) = 4usize, or |c: &Config| c.rpc.as_ref()?.processing_threads,
+			ARG arg_jsonrpc_threads: (Option<usize>) = Some(4), or |c: &Config| c.rpc.as_ref()?.processing_threads,
```

Comment:
If the default is `Some`, should it be `usize` then? How a user can set it to `None` and does it make sense? Maybe we should keep an `Option` and use `unwrap_or(4)` instead?

Reply:
`None` here implies "use defaults", i.e. the user didn't set a value. Either way, for this particular line it's moot: the `jsonrpc-threads` option does nothing and I removed it in https://github.com/paritytech/parity-ethereum/pull/11267/commits/dc0d349d655a44df619558f1bc9f6bb9f8c11c45 – essentially it was once used to set the size of a `CpuPool` but since https://github.com/paritytech/parity-ethereum/pull/9657 it has not had any effect at all.
The equivalent code today sets up the tokio `Executor` which is used for almost all async subsystems (sync, price fetch, rpcs and more) so it is proooobably not something users should be fiddling with. If the need to do so arises we should either use separate executors or introduce some other mechanism of configuring and reserving threads for each subsystem.